lib/swagger-client.js

```diff
@@ -0,0 +1,263 @@
+   * @param {object} options - Options object
+   * @param {string} options.name - Path Item Object name
+   * @param {array} options.splits - Pathname pieces split in '/'
+   * @param {object} options.pathItem - Swagger Path Item Object.
+   */
+  constructor(options) {
+    this.name = options.name;
+    this.splits = options.splits;
+    this.pathItem = options.pathItem;
+  }
+}
+
+class Component {
+  /**
+   * Represents a single path split, child Components, and potentially a Path
+   * Item Object.
+   * @param {object} options - Options object
+   * @param {object} options.http - kubernetes-client Request object
+   * @param {array} options.splits - Absolute pathname (split on '/')
+   * @param {string} options.parameter - Optional Path Template parameter
+   */
+  constructor(options) {
+    let component;
+
+    //
+    // Support Path Templating: use a function to create a a Component-like
+    // object if required. Otherwise use a vanilla object (that isn't callable).
+    //
+    if (options.parameter) {
+      component = function templatedComponent(name) {
+        const splits = component.splits.concat([name]);
+        //
+        // Assume that we'll never have a path with adjacent template parameters.
+        // E.g., assume `/foo/{name}/{property}` cannot exist.
+        //
+        const namedComponent = new Component({
+          splits: splits,
+          http: component.http
+        });
+        component.templatedEndpoints.forEach(child => {
+          namedComponent.addEndpoint(child);
+        });
+        return namedComponent;
+      };
+      component.templatedEndpoints = [];
+
+      //
+      // Attach methods
+      //
+      Object.getOwnPropertyNames(Component.prototype).forEach(name => {
+        if (name === 'constructor') return;
+        component[name] = Component.prototype[name].bind(component);
+      });
+    } else {
+      component = this;
+      component.templatedEndpoints = null;
+    }
+
+    component.parameter = options.parameter;
+    component.splits = options.splits.slice();
+    component.http = options.http;
+    return component;
+  }
+
+  _walkSplits(endpoint) {
+    const splits = this.splits.slice();
+    const nextSplits = endpoint.splits.slice();
+
+    let parent = this;
+    while (nextSplits.length) {
+      const split = nextSplits.shift();
+      splits.push(split);
+
+      let parameter = null;
+      if (nextSplits.length && nextSplits[0].startsWith('{')) {
+        parameter = nextSplits.shift();
+      }
+
+      if (!(split in parent)) {
+        parent[split] = new Component({
+          parameter: parameter,
+          splits: splits,
+          http: this.http
+        });
+      }
+      parent = parent[split];
+
+      //
+      // Path Template parameter: save it and walk it once the user specifies
+      // the value.
+      //
+      if (parameter) {
+        if (!parent.parameter) {
+          throw new Error('Created Component, but require templated one.');
```

Comment:
can we make it a bit more descriptive? like as a developer, what should I do to fix this?